File: yh_divisor/scripts/yh_divisor_pub.py
#!/usr/bin/python
# -*- coding: utf-8 -*-


import rospy
from yh_divisor.msg import yh_divisor_msg

def talker():
   
    pub = rospy.Publisher("yh_divisor_topic",yh_divisor_msg,queue_size=100)
    rospy.init_node("yh_divisor_pub",anonymous=True)
    loop_rate = rospy.Rate(1)
    cnt = 0
    msg = yh_divisor_msg()


    while not rospy.is_shutdown():
        msg.stamp = rospy.get_rostime()
        msg.data = cnt 
        
        cnt += 1
        pub.publish(msg)
        loop_rate.sleep() 
        
if __name__=="__main__":
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

File: yh_divisor/src/yh_divisor_pub.cpp
#include "ros/ros.h"
#include "yh_divisor/yh_divisor_msg.h"

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_divisor_pub");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<yh_divisor::yh_divisor_msg>("yh_divisor_topic", 100);
    ros::Rate loop_rate(1);

    yh_divisor::yh_divisor_msg msg;
    int cnt = 0;

    while(ros::ok())
    {
        msg.stamp = ros::Time::now();
        msg.data = cnt;
        
        cnt++;
        pub.publish(msg);
        loop_rate.sleep();
    }

    return 0;

}